www/js/lib/abstractFilesystemAccess.js

```diff
@@ -88,6 +90,223 @@ StorageFirefoxOS.prototype.enumerate = function (path) {
+ */
+async function handleFolderDropViaFileSystemAPI (packet) {
+    if (!window.params.isFileSystemApiSupported) return true;
+
+    // Only runs when browser support File System API
+    const fileInfo = packet.dataTransfer.items[0];
+    const fileOrDirHandle = await fileInfo.getAsFileSystemHandle();
+    if (fileOrDirHandle.kind === 'file') {
+        localStorage.setItem([fileOrDirHandle.name], [fileOrDirHandle.name].join('|'));
+        cache.idxDB('zimFiles', fileOrDirHandle, function () {
+            // save file in DB
+            updateZimDropdownOptions([fileOrDirHandle.name], fileOrDirHandle.name);
+        });
+        return true;
+    }
+    if (fileOrDirHandle.kind === 'directory') {
+        const fileNames = [];
+        for await (const entry of fileOrDirHandle.values()) {
+            fileNames.push(entry.name);
+        }
+        localStorage.setItem('zimFilenames', fileNames.join('|'));
+        cache.idxDB('zimFiles', fileOrDirHandle, function () {
+            updateZimDropdownOptions(fileNames, '');
+            // save file in DB
+        });
+        return false;
+    }
+}
+
+/**
+ * Handles the folder drop event via WebkitGetAsEntry
+ * @param {DragEvent} event The DragEvent packet
+ * @returns {Promise<{loadZim: boolean, files: Array<File>} | void>} Whether the dropped item is a file or directory and FileList
+ */
+async function handleFolderDropViaWebkit (event) {
+    var dt = event.dataTransfer;
+
+    var entry = dt.items[0].webkitGetAsEntry();
+    if (entry.isFile) {
+        return { loadZim: true, files: [entry.file] };
+    } else if (entry.isDirectory) {
+        var reader = entry.createReader();
+        const files = await getFilesFromReader(reader);
+        const fileNames = [];
+        files.forEach((file) => fileNames.push(file.name));
+        await updateZimDropdownOptions(fileNames, '');
+        return { loadZim: false, files: files };
+    }
+}
+
+/**
+ * Gets the files from the FileSystemReader
+ * @param {FileSystemDirectoryReader} reader The FileSystemReader to get files from
+ * @returns {Promise<Array<File>>} The files from the reader
+ */
+async function getFilesFromReader (reader) {
+    const files = [];
+    const promise = new Promise(function (resolve, _reject) {
+        reader.readEntries(function (entries) {
+            resolve(entries);
+        });
+    });
+    const entries = await promise;
+
+    for (let index = 0; index < entries.length; index++) {
+        const fileOrDir = entries[index];
+        if (fileOrDir.isFile) {
+            const filePromise = await new Promise(function (resolve, _reject) {
+                fileOrDir.file(function (file) {
+                    resolve(file);
+                });
+            });
+            files.push(filePromise);
+        }
+    }
+    return files;
+}
+
 export default {
-    StorageFirefoxOS: StorageFirefoxOS
-};
\ No newline at end of file
+    StorageFirefoxOS: StorageFirefoxOS,
+    updateZimDropdownOptions,
```

Comment:
Throughout the app we use the format `updateZimDropdownOptions: updateZimDropdownOptions,` rather than this shorthand format. For consistency we should probably do the same here. When I was developing the bundling with rollup, I ran into some problems when I tried to use the shorthand format. Maybe those problems have gone away, but I'd feel a bit more reassured if we use the same format throughout. If it can be changed, then we should have a separate issue to change them all.